mk/dev.mk

```diff
@@ -38,32 +38,46 @@ endef
 # so this is location should not be changed by developers.
 KUBECONFIG_DIR := $(HOME)/.kube
 
-PROTOS_DEPS_PATH=$(CI_TOOLS_DIR)/protos
-
-CLANG_FORMAT=$(CI_TOOLS_BIN_DIR)/clang-format
-YQ=$(CI_TOOLS_BIN_DIR)/yq
-HELM=$(CI_TOOLS_BIN_DIR)/helm
-K3D_BIN=$(CI_TOOLS_BIN_DIR)/k3d
-KIND=$(CI_TOOLS_BIN_DIR)/kind
-KUBEBUILDER=$(CI_TOOLS_BIN_DIR)/kubebuilder
-KUBEBUILDER_ASSETS=$(CI_TOOLS_BIN_DIR)
-CONTROLLER_GEN=$(CI_TOOLS_BIN_DIR)/controller-gen
-KUBECTL=$(CI_TOOLS_BIN_DIR)/kubectl
-PROTOC_BIN=$(CI_TOOLS_BIN_DIR)/protoc
-SHELLCHECK=$(CI_TOOLS_BIN_DIR)/shellcheck
-CONTAINER_STRUCTURE_TEST=$(CI_TOOLS_BIN_DIR)/container-structure-test
+ifeq (,$(shell which mise))
+$(error "mise - https://github.com/jdx/mise - not found. Please install it.")
+endif
+MISE := $(shell which mise)
+
+PROTOS_DEPS_PATH=$(shell $(MISE) where protoc)/include
+
+XDS_VERSION=$(shell go list -f '{{ .Version }}' -m github.com/cncf/xds/go)
+PROTO_XDS=$(shell go mod download github.com/cncf/xds@$(XDS_VERSION) && go list -f '{{ .Dir }}' -m github.com/cncf/xds@$(XDS_VERSION))
+PGV_VERSION=$(shell go list -f '{{.Version}}' -m github.com/envoyproxy/protoc-gen-validate)
+PROT_PGV=$(shell go mod download github.com/envoyproxy/protoc-gen-validate@$(PGV_VERSION) && go list -f '{{ .Dir }}' -m github.com/envoyproxy/protoc-gen-validate@$(PGV_VERSION))
+PROTO_GOOGLE_APIS=$(shell go mod download github.com/googleapis/googleapis@master && go list -f '{{ .Dir }}' -m github.com/googleapis/googleapis@master)
+PROTO_ENVOY=$(shell go mod download github.com/envoyproxy/data-plane-api@main && go list -f '{{ .Dir }}' -m github.com/envoyproxy/data-plane-api@main)
+
+CLANG_FORMAT=$(MISE) x clang-format -- clang-format
+YQ=$(shell mise which yq)
+HELM=$(MISE) x helm -- helm
+K3D_BIN=$(MISE) x k3d -- k3d
+KIND=$(MISE) x kind -- kind
+KUBEBUILDER=$(MISE) x kubebuilder -- kubebuilder
+# TODO this should be automated with max k8s version
+KUBEBUILDER_ASSETS_VERSION=1.32.0
+KUBEBUILDER_ASSETS=$(shell setup-envtest use $(KUBEBUILDER_ASSETS_VERSION) --bin-dir $(CI_TOOLS_BIN_DIR) -p path)
```

Comment:
where does this setup-envtest come from? Should it be a mise which?

Reply:
changed to `mise which`